terminal/terminal.md

```diff
@@ -0,0 +1,417 @@
+
+printenv and set (without any arguments) print the defined variables on the session
+
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ printenv
+XDG_SESSION_ID=4
+TERM=xterm-256color
+SHELL=/bin/bash
+SSH_CLIENT=10.0.2.2 50356 22
+SSH_TTY=/dev/pts/0
+USER=vagrant
+```
+and
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ set
+BASH=/bin/bash
+BASHOPTS=checkwinsize:cmdhist:complete_fullquote:expand_aliases:extglob:extquote:force_fignore:histappend:interactive_comments:login_shell:progcomp:promptvars:sourcepath
+BASH_ALIASES=()
+BASH_ARGC=()
+BASH_ARGV=()
+```
+but there's particulary one variable that we want to see and that is **PS1** that controls the look of the shell prompt.
+
+But how do we see what the PS1 variable contains? The command we are looking for to print strings and variables is **echo**, let's try it with always classy hello world
+
+```
+danielvilla@Daniels-MacBook-Air:~$ echo "Hello world"
+Hello world
+```
+now lets see what the PS1 variable has
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ echo $PS1
+\[\e]0;\u@\h: \w\a\]${debian_chroot:+($debian_chroot)}\u@\h:\w\$
+```
+Notice that we had to use **$** sign before the PS1 variable name, it's not used when we inizialiate it, but it is when call it.
+
+Now, yeah, that variable is printed every time you hit enter, that's how we see the prompt, so if we change the value of the $PS1 variable we will see another prompt
+
+```
+export PS1="new prompt"
+```
+Yup, that's your new prompt, to go back just add again the previous value
+
+
+
+#### Other casual commands
+
+These are another handful of useful commands, to give you more vision of your environment while using the terminal, when everything seems dark.
+
+To know what's the user you are currently using
+```
+danielvilla@Daniels-MacBook-Air:~/apprentice$ whoami
+danielvilla
+```
+To know the name of the machine you are currently on
+```
+danielvilla@Daniels-MacBook-Air:~/apprentice$ hostname
+Daniels-MacBook-Air.local
+
+```
+To display a beautiful calendar
+```
+danielvilla@Daniels-MacBook-Air:~/apprentice$ cal
+     March 2018       
+Su Mo Tu We Th Fr Sa  
+             1  2  3  
+ 4  5  6  7  8  9 10  
+11 12 13 14 15 16 17  
+18 19 20 21 22 23 24  
+25 26 27 28 29 30 31  
+```
+To know exactly what day is today with the current time
+```
+danielvilla@Daniels-MacBook-Air:~/apprentice$ date
+Fri Mar  2 02:04:44 MST 2018
+```
+To know the description of a file
+```
+danielvilla@Daniels-MacBook-Air:~/apprentice$ file README.md 
+README.md: UTF-8 Unicode text
+```
+To display a cow
+```
+vagrant@vagrant-ubuntu-trusty-64:~$ apt-get moo
+                 (__) 
+                 (oo) 
+           /------\/ 
+          / |    ||   
+         *  /\---/\ 
+            ~~   ~~   
+..."Have you mooed today?"...
+```
+
+
+#### Asking for help (RTFM)
+
+It's very common to feel lost and don't know how to us a command properly, or for a start what a command do, but don't worry there are some pretty neat commands that help you out you your troubles
```

Comment:
It's very common to feel lost and don't know how to ~us~ `use` a command properly, or for a start what a command do`es`, but don't worry there are some pretty neat commands that help you out you your troubles.
⚠️